staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -48,6 +49,11 @@ var (
 	openapiV3Types                        = sets.NewString("string", "number", "integer", "boolean", "array", "object")
 )
 
+const (
+	// ExpressionCostLimit represents the largest-allowed CEL cost on a per-expression basis.
+	ExpressionCostLimit = 10000000
```

Comment:
it would be helpful to explain the difference between this and `PerCallLimit = 2000000`

Reply:
I could easily see someone using the wrong constant accidentally (I'm not even sure which one is correct to use where)

Reply:
+1, we should try to clearly distinguish a couple things:

This is an "estimated" cost. It's "per-rule" (each rule defined in a CRD must not exceed this limit). It accounts for the fact that a rule can be called many times (i.e. if it's within a map or array).

`PerCallLimit` on the other hand, is an "actual" cost limit for the cost exhausted at runtime evaluating a rule. And it's a per-call limit.

Reply:
Right - the constant is now `StaticEstimatedCostLimit` to ensure no confusion with the runtime limit.